org.alloytools.alloy.cli/src/main/java/org/alloytools/alloy/cli/CLI.java

```diff
@@ -0,0 +1,145 @@
+package org.alloytools.alloy.cli;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.alloytools.alloy.infrastructure.api.AlloyMain;
+
+import aQute.lib.env.Env;
+import aQute.lib.getopt.Arguments;
+import aQute.lib.getopt.Description;
+import aQute.lib.getopt.Options;
+import aQute.lib.io.IO;
+import aQute.lib.json.JSONCodec;
+import aQute.libg.glob.Glob;
+import edu.mit.csail.sdg.ast.Command;
+import edu.mit.csail.sdg.parser.CompModule;
+import edu.mit.csail.sdg.parser.CompUtil;
+import edu.mit.csail.sdg.translator.A4Options;
+import edu.mit.csail.sdg.translator.A4Solution;
+import edu.mit.csail.sdg.translator.A4SolutionWriter;
+import edu.mit.csail.sdg.translator.TranslateAlloyToKodkod;
+
+@AlloyMain(name = { "exec" })
+public class CLI extends Env {
+	final A4Options		options	= new A4Options();
+
+	public enum OutputType {
+		plain, json, xml;
+	}
+
+	/**
+	 * Show the list of solvers
+	 */
+
+	/**
+	 * Run all 'run' & asserts
+	 */
+
+	@Arguments(arg = "path")
+	@Description("Execute an Alloy program")
+	interface ExecOptions extends Options {
+		@Description("The command to run. If no command is specified, the default command will run. The command may specify wildcards")
+		String command();
+
+		@Description("Specify the output type: plain, json, xml")
+		OutputType type(OutputType deflt);
+
+		@Description("Specify where the output should go. Default is the console")
+		String output();
+		
+		@Description("If set, forbids overflow from occurring")
+		boolean nooverflow();
+		
+		@Description("Number of allowed recursion unrolls. Default is -1, which indicates no unrolls")
+		int unrolls(int n);
+		@Description("Depth for skolem analysis, default is 0")
+		int depth(int n);
+		
+		@Description("Symmetry breaking")
```

Comment:
I think this needs to be more descriptive and give some intuition about what difference it makes